datafusion/physical-expr/src/intervals/interval_aritmetic.rs

```diff
@@ -451,6 +453,75 @@ impl Interval {
         lower: IntervalBound::new(ScalarValue::Boolean(Some(true)), false),
         upper: IntervalBound::new(ScalarValue::Boolean(Some(true)), false),
     };
+
+    // Cardinality is the number of all points included by the interval, considering its bounds.
```

Comment:
Something that might be worth considering in the long term that @tustvold mentioned the other day is to vectorize these calculations -- at the moment they are doin in the context of a single expression, but eventually if we want to use this logic to prune large numbers of files / etc based on statistics it may take too long

No change is needed here, I am just planting a seed of an idea in case this was on your list too

Reply:
Yes, you are right. Speaking of this idea, I also want to mention that we have tried to replace the pruning logic in PruningPredicate with this interval library but realized that it disrupts the vectorized calculations. So I'm thinking about how we can use it without breaking the vectorized process.